### PR TITLE
[MSVC] Use string view literals with StringIndexer

### PR DESCRIPTION
**Pull request by mwilsnd**
When building with MSVC, a linker issue can be observed with `StringIndexer::get`. This is apparently caused by string literals above the string SSO size failing to implicitly convert to `std::string_view`s. This PR brings `std::string_view_literals` in with `string_indexer.hpp` and calls to `get` with string literals are explicitly provided as string view literals (`"string"sv).

### CODE REVIEW

**Review by sjg-wdw — APPROVED** [review]
I'm glad you're doing real time integration on Windows.  This would be an incredible headache later.

**Comment by TimSylvester** [comment]
It seems likely that new code might fail to include `sv`.  Would the implicit conversion work better if we removed the bare `char*` overload?  Or at least produce a compile issue, thus forcing it on other platforms?

**Comment by mwilsnd** [comment]
Is there a `char*` overload? I don't remember seeing one. I do agree it's likely the explicit `sv` will be forgotten sooner or later.

**Comment by TimSylvester** [comment]
There was, but it looks like it already got removed.  So... maybe add it back to make the conversion explicit?  Or add `explicit` to the argument?

https://github.com/maplibre/maplibre-native/pull/1644/files

File: include/mbgl/util/string_indexer.hpp [hunk 1]
```diff
@@ -9,6 +9,8 @@
 #include <mutex>
 #include <shared_mutex>
 
+using namespace std::string_view_literals;
```
Comment (TimSylvester) [inline include/mbgl/util/string_indexer.hpp:12]: Better not to have this in the global namespace, but at least within `mbgl` or not in the header at all.

**Review by TimSylvester — APPROVED** [review]

**Comment by mwilsnd** [comment]
Well today I can't reproduce the linker issue, closing this for now.

**Comment by stefankarschti** [comment]
we could move from `string_view` to `const char*` parameter type and a few overloads to accept `std::string` / `string_view` maybe? I think this would work out of the box...

**Pull request closed without merging**